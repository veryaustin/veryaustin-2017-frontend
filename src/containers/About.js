import React, { Component, PropTypes } from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { fetchPage } from "../actions/fetchActions";
import Banner from "../components/Banner";
import Section from "../components/Section";
import Loading from "../components/Loading";

class About extends Component {
  componentDidMount() {
    this.props.fetchPage(this.props.route.path);
  }

  render() {
    // Destructure the about object & loading/error props
    const {
      about: {
        title,
        caption,
        image_url,
        banner_url,
        content,
        resume_site_url,
        resume_download_url,
        tools
      },
      isLoading,
      hasErrored
    } = this.props;

    return (
      <div>
        {isLoading ? (
          <Loading />
        ) : (
          <div>
            <Banner
              title={title}
              caption={caption}
              featured_image={banner_url}
              dark
            />
            <Section
              description={content}
              image_url={image_url}
              items={tools}
              primary_site_url={resume_site_url}
              secondary_site_url={resume_download_url}
              label={"Tools Of The Trade:"}
              primary_site_button={"View Resume Site"}
              secondary_site_button={"Download Resume"}
            />
          </div>
        )}
      </div>
    );
  }
}

// Map State To Props
const mapStateToProps = ({
  pages: { about },
  fetching: { isLoading, error }
}) => {
  return {
    about,
    isLoading,
    hasErrored: error
  };
};

// Bind the actions and dispatch them
const mapDispatchToProps = dispatch => {
  return bindActionCreators({ fetchPage }, dispatch);
};

// About Props Validation
About.propTypes = {
  fetchPage: PropTypes.func,
  isLoading: PropTypes.bool,
  hasErrored: PropTypes.string,
  about: PropTypes.object,
  route: PropTypes.object
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(About);
