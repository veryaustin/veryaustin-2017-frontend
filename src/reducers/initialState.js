export default {
  pages: {
    home: {},
    work: {},
    about: {},
    contact: {}
  },
  fetching: {
    isLoading: true,
    error: null
  }
};
