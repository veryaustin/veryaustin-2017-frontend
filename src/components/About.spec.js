import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import About from './About';
import initialState from '../reducers/initialState';

describe('About Component', () => {
 let middlewares, mockStore, store;
 beforeEach(() => {
  middlewares = [];
  mockStore = configureStore(middlewares);
  store = mockStore(initialState);
 });

 it('renders the about component', () => {
    const wrapper = shallow(<About store={store} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});