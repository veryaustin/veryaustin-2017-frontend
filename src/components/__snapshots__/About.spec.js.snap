exports[`About Component renders the about component 1`] = `
<About
  about={Object {}}
  fetchAbout={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  } />
`;
